src/test/java/com/artipie/npm/proxy/NpmProxyTest.java

```diff
@@ -0,0 +1,305 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 artipie
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie.npm.proxy;
+
+import com.artipie.asto.Concatenation;
+import com.artipie.asto.Key;
+import com.artipie.asto.Storage;
+import com.artipie.asto.blocking.BlockingStorage;
+import com.artipie.asto.memory.InMemoryStorage;
+import com.artipie.npm.proxy.model.NpmAsset;
+import com.artipie.npm.proxy.model.NpmPackage;
+import io.vertx.junit5.VertxExtension;
+import io.vertx.junit5.VertxTestContext;
+import io.vertx.reactivex.core.Vertx;
+import io.vertx.reactivex.core.http.HttpServer;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.TimeUnit;
+import net.minidev.json.JSONObject;
+import net.minidev.json.parser.JSONParser;
+import net.minidev.json.parser.ParseException;
+import org.apache.commons.configuration2.BaseConfiguration;
+import org.apache.commons.io.IOUtils;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.core.IsEqual;
+import org.json.JSONException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.skyscreamer.jsonassert.JSONAssert;
+
+/**
+ * Test NPM Proxy works.
+ *
+ * @since 0.1
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ */
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
+@ExtendWith(VertxExtension.class)
+final class NpmProxyTest {
```

Comment:
@brastak I still do not completely understand whether or not are we trying to write code in EO style or not, also if `NpmProxy` will be refactored, then this test has to be refactored too... But just in case here are several articles about best practice in tests: 
[Test method names](https://www.yegor256.com/2014/04/27/typical-mistakes-in-java-code.html#test-method-names)
[Single Statement Unit Tests](https://www.yegor256.com/2017/05/17/single-statement-unit-tests.html)
[Unti patterns](https://www.yegor256.com/2018/12/11/unit-testing-anti-patterns.html)

Reply:
Hm, I don't know what I need to answer ) Ok, I agree that if NpmProxy will be divided, this test should be divided too.
I don't think that this test should be called Unit test - it is integration test by it nature. But I have seen several levels of abstraction in our tests (in the other projects). We used real repository clients in integration tests (with docker environment). These tests are in the higher level. These tests are heavier. NpmProxyTest is designed to be more lightweight and to be executed quickly. It produces in-memory side-effects only. That's why it's placed with unit tests. It checks both API responses and all side effects as white-box and I scare about the idea that all these checks should be rewrited in the single-assert approach.
Regarding naming - ok, I will rename it, but it's really just the work for a work. There are dozens  wide-used naming conventions. I can explain why I use mine (it just supported from the earliest versions of JUnit). And what about your links? )

Reply:
@brastak in my experience it's difficult to write tests for a lot of people, to maintain them up to date is even more harder. EO suggestions about test methods naming and single assertion per test method rule help a lot.

Reply:
In general, I can agree with you. This approach is useful when we have a deal with simple cases. But sometimes the logic is complex enough to use simple method names (usually, when you need to test some decision tree). In this cases "given_when_then" naming convention can be useful. It looks ugly, but it really helps to find a specific case. And sometimes your checks depend on several states, not the return value only, but also you want to verify some side-effects (like in this test). In this case it's impossible to use just one assert. Of course, we can write two tests for one call - the first will check return value and the second - side-effect (storage state, in this case). But it looks like redundant